apps/ledger-live-mobile/src/screens/PairDevices/Scanning.tsx

```diff
@@ -18,18 +18,35 @@ type Props = {
   onSelect: (device: Device, deviceMeta: any) => Promise<void>;
   onError: (_: Error) => void;
   onTimeout: () => void;
+  /** If defined, only show devices that have a device model id in this array */
+  deviceModelIds?: DeviceModelId[];
 };
 
 type Device = {
   id: string;
   name: string;
 };
 
-export default function Scanning({ onTimeout, onError, onSelect }: Props) {
+export default function Scanning({
+  onTimeout,
+  onError,
+  onSelect,
+  deviceModelIds,
+}: Props) {
   const { t } = useTranslation();
   const knownDevices = useSelector(knownDevicesSelector);
   const [devices, setDevices] = useState<Device[]>([]);
 
+  const filteredDevices = useMemo(() => {
+    if (!deviceModelIds) return devices;
+    return devices.filter(device => {
+      let modelId = "nanoX" as DeviceModelId;
+      const infos = getInfosForServiceUuid(device.serviceUUIDs[0]);
+      if (infos) modelId = infos.deviceModel.id;
+      return deviceModelIds.includes(modelId);
+    });
+  }, [devices, deviceModelIds]);
```

Comment:
it's cool - but just for your info, the new BLE pairing flow has this filtering on device model ids already 🙈 and this Scanning screen will no longer be used

Reply:
See `apps/ledger-live-mobile/src/screens/BleDevicePairingFlow/BleDevicesScanning.tsx`